### PR TITLE
:bug: Fix denops error when call component update before plugin startup

### LINKED ISSUE #73: Unexpectedly denops error when call component at before plugin startup and fire BufEnter.
# Problem

Unexpectedly denops error when call component at before plugin startup and fire BufEnter.

# To Reproduce

execute Vim with following vimrc.

```vim
set rtp+=/data/vim/repos/github.com/vim-denops/denops.vim
set rtp+=/data/vim/repos/github.com/lambdalisue/gin.vim

let s:pattern = 1
if s:pattern == 1
  call gin#component#branch#ascii()
endif
if s:pattern == 2
  au VimEnter * call gin#component#branch#ascii()
  au VimEnter * ++nested e hoge
endif
```

after print this error.

```
Error detected while processing command line..BufEnter Autocommands for "*"..function gin#internal#component#update[2]..denops#request_async[7]..denops#server#request[4]..denops#_internal#server#chan#request[2]..command line..BufEnter Autocommands for "*"..function gin#internal#component#update[2]..denops#request_async[7]..denops#server#request[4]..denops#_internal#server#chan#request:
line    2:
E605: Exception not caught: [denops] Channel is not ready yet
```

# Note

This problem is caused by a hook in BufEnter that is called before the plugin is started.

https://github.com/lambdalisue/gin.vim/blob/e7a6bd489b3427df66bcd755dc291544f09372da/autoload/gin/internal/component.vim#L11

### PR DESCRIPTION
**Pull request by kuuote**
Plan of fix #73
Add plugin loaded check at update.

### CODE REVIEW

**Comment by kuuote** [comment]
It adds force update at initialize because can call update anytime now.
Should I divide it to another PR?

File: autoload/gin/internal/component.vim [hunk 1]
```diff
@@ -12,13 +12,17 @@ function! gin#internal#component#init(component) abort
     execute printf('autocmd User GinCommandPost call gin#internal#component#update("%s")', a:component)
     execute printf('autocmd User DenopsPluginPost:gin call gin#internal#component#update("%s")', a:component)
   augroup END
+  call gin#internal#component#update(a:component)
```
Comment (lambdalisue) [inline autoload/gin/internal/component.vim:15]: Why do we need this?

**Review by lambdalisue — CHANGES_REQUESTED** [review]
> It adds force update at initialize because can call update anytime now.
Should I divide it to another PR?

I don't get the reason so please divide PRs.

File: autoload/gin/internal/component.vim (hunk 1 above)
Comment (kuuote) [reply autoload/gin/internal/component.vim:15]: It fix didn't call update until next BufEnter when called after plugin initialized.

Comment (kuuote) [reply autoload/gin/internal/component.vim:15]: Will can call this timing at this fix, so I did, but I thought shouldn't mixed it. delete from this PR.

**Review by lambdalisue — APPROVED** [review]

**Pull request merged**